### PR TITLE
Scrollable comments

### LINKED ISSUE #1870: Comment scroll
**Describe the new addon**
Some people compose very lengthy comments. Even with the 500-word limit, it is frustrating to read that lengthy comment while scrolling the entire page. This addon will make the comment div scrollable so that the whole page is not scrolled down.

### PR DESCRIPTION
**Pull request by Shluffy**
Resolves #1870 

### Changes

adds some css to make them scroll

### Reason for changes

because there is an open issue, and a good addon idea

### Tests

none. see below though

## Notes
This is not my code. I properly formatted the code from pr #2744. I am not in the credits section for that reason

### CODE REVIEW

**Comment by Shluffy** [comment]
cc @Genius398

**Comment by Genius398** [comment]
Did you test it?

**Comment by Shluffy** [comment]
no, see the test part of the template

File: addons/comment-scroll/addon.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+      "url": "userstyle.css",
+      "matches": [
+        "https://scratch.mit.edu/users/*",
+        "https://scratch.mit.edu/projects/*",
+        "https://scratch.mit.edu/studios/*"
+      ]
+    }
+  ],
+  "tags": ["comments", "community"],
+  "enabled_by_default": false
```
Comment (FunctionalMetatable) [inline addons/comment-scroll/addon.json:21]: Should be `enabledByDefault`.

e.g
https://github.com/ScratchAddons/ScratchAddons/blob/9afb7dc71684b42b3bc4ca8d4b6967ed9409849d/addons/remove-sprite-confirm/addon.json#L12

File: addons/comment-scroll/addon.json [hunk 2]
```diff
@@ -0,0 +1,22 @@
+{
+  "name": "Comment Scroll",
+  "description": "Makes Comments On the scratch website scrollable, which can reduce the height of the page.",
```
Comment (FunctionalMetatable) [inline addons/comment-scroll/addon.json:3]: Sentence case for descriptions!

**Review by FunctionalMetatable — CHANGES_REQUESTED** [review]

File: addons/comment-scroll/addon.json (hunk 1 above)
Comment (Shluffy) [reply addons/comment-scroll/addon.json:21]: I didn't write this, i'll fix though

**Comment by FunctionalMetatable** [comment]
Is it ok if I merge this into another PR which merges both of these redirects? I'll add you as credit @Shluffy

File: addons/comment-scroll/addon.json [hunk 3]
```diff
@@ -0,0 +1,22 @@
+{
+  "name": "Comment Scroll",
+  "description": "Makes comments on the Scratch website scrollable, which can reduce the height of the page.",
```
Comment (RedGuy12) [inline addons/comment-scroll/addon.json:3]:
```suggestion
  "name": "Scrollable comments",
  "description": "Makes comments scrollable, which can reduce the height of the page.",
```

File: addons/addons.json [hunk 4]
```diff
@@ -80,6 +80,7 @@
   "redirect-mobile-forums",
   "expanding-search-bar",
   "debugger",
+  "comment-scroll"
```
Comment (WorldLanguages) [inline addons/addons.json:83]: This is invalid JSON you forgot the comma here

**Comment by Daniel4-Scratch** [comment]
I just have a question. Is this **really** needed?

**Comment by TheColaber** [comment]
I think this is only useful if you have the linebreak addon enabled... other than that it's useless.

Also, this 100% was not tested at all. I'm surprised no one else did either.
image
image
Pls close this and find a way to merge this with the linebreak addon, and actually TEST this. Yes I realize this was originally by @Genius398 so they may create a PR if needed.

**Comment by Daniel4-Scratch** [comment]
Maybe it should be enabled when the line breaker is enabled. So it could be in the line breaker code.

**Review by Daniel4-Scratch — CHANGES_REQUESTED** [review]
In the user styles you should add `"https://scratch.mit.edu/messages"`.
Because comments also show in the messages page.

File: addons/comment-scroll/addon.json [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+  "userstyles": [
+    {
+      "url": "userstyle.css",
+      "matches": [
+        "https://scratch.mit.edu/users/*",
+        "https://scratch.mit.edu/projects/*",
+        "https://scratch.mit.edu/studios/*"
+      ]
+    }
+  ],
```
Comment (FunctionalMetatable) [inline addons/comment-scroll/addon.json:19]: You should rewrite the CSS differently for scratch-www and scratchr2. Also, add yourself into the credits section as you PR'd it

**Comment by Shluffy** [comment]
Okay. I am going to close this as it seems to have to many problems. @Genius398 feel free to test it, and make a new one that works.

**Comment by WorldLanguages** [comment]
Keep in mind that a working version might not get accepted, because whether this is useful is up to debate

**Comment by Shluffy** [comment]
i don't really care, as I said this ins't my code

**Comment by Genius398** [comment]
Ive been mentioned 1000 times here without noticing

**Pull request closed without merging**